<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.549">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Research Reference - About</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<meta property="og:title" content="Research Reference - About">
<meta property="og:description" content="">
<meta property="og:site_name" content="Research Reference">
<meta name="citation_title" content="About">
<meta name="citation_fulltext_html_url" content="ref.drpranavmishra.com/about.html">
<meta name="citation_language" content="en-US">
<meta name="citation_reference" content="citation_title=Protocol for making a 4% formaldehyde solution in PBS. R&amp;amp;amp;d systems;,citation_abstract=Read our Protocol for Making a 4% Formaldehyde Solution in PBS.;,citation_author=R&amp;amp;D Systems;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.rndsystems.com/resources/protocols/protocol-making-4-formaldehyde-solution-pbs;,citation_language=en-US;">
<meta name="citation_reference" content="citation_title=Phosphate-buffered saline (PBS);,citation_publication_date=2006-06;,citation_cover_date=2006-06;,citation_year=2006;,citation_fulltext_html_url=http://www.cshprotocols.org/lookup/doi/10.1101/pdb.rec8247;,citation_issue=1;,citation_doi=10.1101/pdb.rec8247;,citation_issn=1940-3402, 1559-6095, 1559-6095;,citation_volume=2006;,citation_language=en-US;,citation_journal_title=Cold Spring Harbor Protocols;,citation_journal_abbrev=Cold Spring Harb Protoc;">
<meta name="citation_reference" content="citation_title=Transfer and staining of proteins in western blot. Abcam;,citation_author=Abcam;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.abcam.com/protocols/transfer-and-staining-of-proteins-in-western-blot;">
<meta name="citation_reference" content="citation_title=Western blot protocols and recipes. ThermoFicher scientific;,citation_abstract=Find SDS page protocols and western blot protocols for every step of the workflow, including common electrophoresis recipes and western blot buffer recipes and materials.;,citation_author=ThermoFisher;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.thermofisher.com/us/en/home/life-science/protein-biology/protein-biology-learning-center/protein-gel-electrophoresis-information/western-blot-protocols.html;,citation_language=en-US;">
<meta name="citation_reference" content="citation_title=General protocol for western blotting;,citation_author=Bio-Rad;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.bio-rad.com/webroot/web/pdf/lsr/literature/Bulletin_6376.pdf;,citation_publisher=Bio-Rad Laboratories, Inc;">
<meta name="citation_reference" content="citation_title=Western blotting - a beginner’s guide;,citation_author=Abcam;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.abcam.com/ps/products/131/ab131366/documents/Western%20blotting%20-%20a%20beginner&amp;amp;amp;#039;s%20guide.pdf;,citation_publisher=Abcam;">
<meta name="citation_reference" content="citation_title=Western blot buffers and stock solutions. Abcam;,citation_author=Abcam;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.abcam.com/protocols/buffer-and-stock-solutions-for-western-blot#TBS-10x;">
<meta name="citation_reference" content="citation_title=Levene’s test (statistics) - the ultimate guide. SPSS tutorials;,citation_abstract=Levene’s test evaluates the homogeneity assumption required by ANOVA and t-tests: do all groups have equal population variances?;,citation_author=Ruben Geert Berg;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.spss-tutorials.com/levenes-test-in-spss/;,citation_language=en-US;">
<meta name="citation_reference" content="citation_title=Colonic epithelial cell diversity in health and inflammatory bowel disease;,citation_author=Kaushal Parikh;,citation_author=Agne Antanaviciute;,citation_author=David Fawkner-Corbett;,citation_author=Marta Jagielowicz;,citation_author=Anna Aulicino;,citation_author=Christoffer Lagerholm;,citation_author=Simon Davis;,citation_author=James Kinchen;,citation_author=Hannah H. Chen;,citation_author=Nasullah Khalid Alham;,citation_author=Neil Ashley;,citation_author=Errin Johnson;,citation_author=Philip Hublitz;,citation_author=Leyuan Bao;,citation_author=Joanna Lukomska;,citation_author=Rajinder Singh Andev;,citation_author=Elisabet Björklund;,citation_author=Benedikt M. Kessler;,citation_author=Roman Fischer;,citation_author=Robert Goldin;,citation_author=Hashem Koohy;,citation_author=Alison Simmons;,citation_publication_date=2019-03-07;,citation_cover_date=2019-03-07;,citation_year=2019;,citation_fulltext_html_url=https://www.nature.com/articles/s41586-019-0992-y;,citation_issue=7746;,citation_doi=10.1038/s41586-019-0992-y;,citation_issn=0028-0836, 1476-4687;,citation_volume=567;,citation_language=en-US;,citation_journal_title=Nature;,citation_journal_abbrev=Nature;">
<meta name="citation_reference" content="citation_title=Tissue prep - official 10x genomics support. 10x genomics;,citation_fulltext_html_url=https://www.10xgenomics.com/support/spatial-gene-expression-ffpe/documentation/workflows/ffpe-v-1/steps/tissue-prep/visium-spatial-gene-expression-for-ffpe-tissue-preparation-guide;,citation_language=en-US;">
<meta name="citation_reference" content="citation_title=Spatiotemporal analysis of human intestinal development at single-cell resolution;,citation_author=David Fawkner-Corbett;,citation_author=Agne Antanaviciute;,citation_author=Kaushal Parikh;,citation_author=Marta Jagielowicz;,citation_author=Ana Sousa Gerós;,citation_author=Tarun Gupta;,citation_author=Neil Ashley;,citation_author=Doran Khamis;,citation_author=Darren Fowler;,citation_author=Edward Morrissey;,citation_author=Chris Cunningham;,citation_author=Paul R. V. Johnson;,citation_author=Hashem Koohy;,citation_author=Alison Simmons;,citation_publication_date=2021-02;,citation_cover_date=2021-02;,citation_year=2021;,citation_fulltext_html_url=https://linkinghub.elsevier.com/retrieve/pii/S009286742031686X;,citation_issue=3;,citation_doi=10.1016/j.cell.2020.12.016;,citation_issn=00928674;,citation_volume=184;,citation_language=en-US;,citation_journal_title=Cell;,citation_journal_abbrev=Cell;">
<meta name="citation_reference" content="citation_title=Making formalin-fixed, paraffin embedded blocks;,citation_abstract=Paraffin embedding is a standard technique used in clinical and research laboratories to create a formalin-fixed, paraffin-embedded (FFPE) block of tissue. Formalin-fixed tissue undergoes tissue processing and then is embedded in paraffin (wax) to create a FFPE block or paraffin block. The paraffin block can be cut using a microtome to generate thin sections of tissue contained in paraffin to be stained or paraffin tissue ribbons suitable for nucleic acid extraction. In addition, the FFPE blocks can be stored at room temperature for years. Herein, we provide a basic knowledge, and introduce common methods of the paraffin embedding process.;,citation_author=Alireza Sadeghipour;,citation_author=Pegah Babaheidarian;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_doi=10.1007/978-1-4939-8935-5_22;,citation_issn=1940-6029;,citation_pmid=30539450;,citation_volume=1897;,citation_journal_title=Methods in Molecular Biology (Clifton, N.J.);,citation_journal_abbrev=Methods Mol Biol;">
<meta name="citation_reference" content="citation_title=Museum of spatial transcriptomics;,citation_abstract=The function of many biological systems, such as embryos, liver lobules, intestinal villi, and tumors, depends on the spatial organization of their cells. In the past decade, high-throughput technologies have been developed to quantify gene expression in space, and computational methods have been developed that leverage spatial gene expression data to identify genes with spatial patterns and to delineate neighborhoods within tissues. To comprehensively document spatial gene expression technologies and data-analysis methods, we present a curated review of literature on spatial transcriptomics dating back to 1987, along with a thorough analysis of trends in the field, such as usage of experimental techniques, species, tissues studied, and computational approaches used. Our Review places current methods in a historical context, and we derive insights about the field that can guide current research strategies. A companion supplement offers a more detailed look at the technologies and methods analyzed: https://pachterlab.github.io/LP_2021/.;,citation_author=Lambda Moses;,citation_author=Lior Pachter;,citation_publication_date=2022-05;,citation_cover_date=2022-05;,citation_year=2022;,citation_fulltext_html_url=https://www.nature.com/articles/s41592-022-01409-2;,citation_issue=5;,citation_doi=10.1038/s41592-022-01409-2;,citation_issn=1548-7105;,citation_volume=19;,citation_language=en-US;,citation_journal_title=Nature Methods;,citation_journal_abbrev=Nat Methods;">
<meta name="citation_reference" content="citation_title=Spatially resolved transcriptomic profiling of degraded and challenging fresh frozen samples;,citation_abstract=Spatially resolved transcriptomics has enabled precise genome-wide mRNA expression profiling within tissue sections. The performance of methods targeting the polyA tails of mRNA relies on the availability of specimens with high RNA quality. Moreover, the high cost of currently available spatial resolved transcriptomics assays requires a careful sample screening process to increase the chance of obtaining high-quality data. Indeed, the upfront analysis of RNA quality can show considerable variability due to sample handling, storage, and/or intrinsic factors. We present RNA-Rescue Spatial Transcriptomics (RRST), a workflow designed to improve mRNA recovery from fresh frozen specimens with moderate to low RNA quality. First, we provide a benchmark of RRST against the standard Visium spatial gene expression protocol on high RNA quality samples represented by mouse brain and prostate cancer samples. Then, we test the RRST protocol on tissue sections collected from five challenging tissue types, including human lung, colon, small intestine, pediatric brain tumor, and mouse bone/cartilage. In total, we analyze 52 tissue sections and demonstrate that RRST is a versatile, powerful, and reproducible protocol for fresh frozen specimens of different qualities and origins.;,citation_author=Reza Mirzazadeh;,citation_author=Zaneta Andrusivova;,citation_author=Ludvig Larsson;,citation_author=Phillip T. Newton;,citation_author=Leire Alonso Galicia;,citation_author=Xesús M. Abalo;,citation_author=Mahtab Avijgan;,citation_author=Linda Kvastad;,citation_author=Alexandre Denadai-Souza;,citation_author=Nathalie Stakenborg;,citation_author=Alexandra B. Firsova;,citation_author=Alia Shamikh;,citation_author=Aleksandra Jurek;,citation_author=Niklas Schultz;,citation_author=Monica Nistér;,citation_author=Christos Samakovlis;,citation_author=Guy Boeckxstaens;,citation_author=Joakim Lundeberg;,citation_publication_date=2023-01-31;,citation_cover_date=2023-01-31;,citation_year=2023;,citation_issue=1;,citation_doi=10.1038/s41467-023-36071-5;,citation_issn=2041-1723;,citation_pmid=36720873;,citation_volume=14;,citation_journal_title=Nature Communications;,citation_journal_abbrev=Nat Commun;">
<meta name="citation_reference" content="citation_title=Degradation of intestinal mRNA: A matter of treatment;,citation_abstract=AIM: To characterize the influence of location, species and treatment upon RNA degradation in tissue samples from the gastrointestinal tract., METHODS: The intestinal samples were stored in different medium for different times under varying conditions: different species (human and rat), varying temperature (storage on crushed ice or room temperature), time point of dissection of the submucous-mucous layer from the smooth muscle (before or after storage), different rinsing methods (rinsing with Medium, PBS, RNALater or without rinsing at all) and different regions of the gut (proximal and distal small intestine, caecum, colon and rectum). The total RNA from different parts of the gut (rat: proximal and distal small intestine, caecum, colon and rectum, human: colon and rectum) and individual gut layers (muscle and submucosal/mucosal) was extracted. The quality of the RNA was assessed by micro capillary electrophoresis. The RNA quality was expressed by the RNA integrity number which is calculated from the relative height and area of the 18 S and 28 S RNA peaks. From rat distal small intestine qPCR was performed for neuronal and glial markers., RESULTS: RNA obtained from smooth muscle tissue is much longer stable than those from submucosal/mucosal tissue. At RT muscle RNA degrades after one day, on ice it is stable at least three days. Cleaning and separation of gut layers before storage and use of RNALater, maintains the stability of muscle RNA at RT for much longer periods. Different parts of the gut show varying degradation periods. RNA obtained from the submucosal/mucosal layer always showed a much worse amplification rate than RNA from muscle tissue. In general RNA harvested from rat tissue, either smooth muscle layer or submucosal/mucosal layer is much longer stable than RNA from human gut tissue, and RNA obtained from smooth muscle tissue shows an increased stability compared to RNA from submucosal/mucosal tissue. At RT muscle RNA degrades after one day, while the stability on ice lasts at least three days. Cleaning and separation of gut layers before storage and use of RNALater, maintains the stability of muscle RNA at RT for much longer periods. Different parts of the gut show varying degradation periods. The RNA from muscle and submucosal/mucosal tissue of the proximal small intestine degrades much faster than the RNA of distal small intestine, caecum or colon with rectum. RNA obtained from the submucosal/mucosal layer always showed a much more reduced amplification rate than RNA from muscle tissue [β-Tubulin III for muscle quantification cycle (Cp): 22.07 ± 0.25, for β-Tubulin III submucosal/mucosal Cp: 27.42 ± 0.19]., CONCLUSION: Degradation of intestinal mRNA depends on preparation and storage conditions of the tissue. Cooling, rinsing and separating of intestinal tissue reduce the degradation of mRNA.;,citation_author=Sabine Heumüller-Klug;,citation_author=Carsten Sticht;,citation_author=Karin Kaiser;,citation_author=Elvira Wink;,citation_author=Cornelia Hagl;,citation_author=Lucas Wessel;,citation_author=Karl-Herbert Schäfer;,citation_publication_date=2015-03-28;,citation_cover_date=2015-03-28;,citation_year=2015;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4375571/;,citation_issue=12;,citation_doi=10.3748/wjg.v21.i12.3499;,citation_issn=1007-9327;,citation_pmid=25834314;,citation_volume=21;,citation_journal_title=World Journal of Gastroenterology : WJG;,citation_journal_abbrev=World J Gastroenterol;">
<meta name="citation_reference" content="citation_title=An introduction to spatial transcriptomics for biomedical research;,citation_abstract=Single-cell transcriptomics (scRNA-seq) has become essential for biomedical research over the past decade, particularly in developmental biology, cancer, immunology, and neuroscience. Most commercially available scRNA-seq protocols require cells to be recovered intact and viable from tissue. This has precluded many cell types from study and largely destroys the spatial context that could otherwise inform analyses of cell identity and function. An increasing number of commercially available platforms now facilitate spatially resolved, high-dimensional assessment of gene transcription, known as “spatial transcriptomics”. Here, we introduce different classes of method, which either record the locations of hybridized mRNA molecules in tissue, image the positions of cells themselves prior to assessment, or employ spatial arrays of mRNA probes of pre-determined location. We review sizes of tissue area that can be assessed, their spatial resolution, and the number and types of genes that can be profiled. We discuss if tissue preservation influences choice of platform, and provide guidance on whether specific platforms may be better suited to discovery screens or hypothesis testing. Finally, we introduce bioinformatic methods for analysing spatial transcriptomic data, including pre-processing, integration with existing scRNA-seq data, and inference of cell-cell interactions. Spatial -omics methods are already improving our understanding of human tissues in research, diagnostic, and therapeutic settings. To build upon these recent advancements, we provide entry-level guidance for those seeking to employ spatial transcriptomics in their own biomedical research.;,citation_author=Cameron G. Williams;,citation_author=Hyun Jae Lee;,citation_author=Takahiro Asatsuma;,citation_author=Roser Vento-Tormo;,citation_author=Ashraful Haque;,citation_publication_date=2022-06-27;,citation_cover_date=2022-06-27;,citation_year=2022;,citation_fulltext_html_url=https://doi.org/10.1186/s13073-022-01075-1;,citation_issue=1;,citation_doi=10.1186/s13073-022-01075-1;,citation_issn=1756-994X;,citation_volume=14;,citation_journal_title=Genome Medicine;,citation_journal_abbrev=Genome Medicine;">
<meta name="citation_reference" content="citation_title=Comparison of fresh frozen tissue with formalin-fixed paraffin-embedded tissue for mutation analysis using a multi-gene panel in patients with colorectal cancer;,citation_abstract=Background: Next generation sequencing (NGS)-based multi-gene panel tests have been performed to predict the treatment response and prognosis in patients with colorectal cancer (CRC). Whether the multi-gene mutation results of formalin-fixed paraffin-embedded (FFPE) tissues are identical to those of fresh frozen tissues remains unknown., Methods: A 22-gene panel with 103 hotspots was used to detect mutations in paired fresh frozen tissue and FFPE tissue from 118 patients with CRC., Results: In our study, 117 patients (99.2%) had one or more variants, with 226 variants in FFPE tissue and 221 in fresh frozen tissue. Of the 129 variants identified in this study, 96 variants were present in both FFPE and fresh frozen tissues; 27 variants were found in FFPE tissues only; 6 variants were found only in fresh frozen tissues. The mutation results demonstrated &amp;amp;amp;gt;94.0% concordance in all variants, with Kappa coefficient &amp;gt;0.500 in 64.3% (83/129) of variants. At the gene level, concordance ranged from 73.8 to 100.0%, with Kappa coefficient >0.500 in 81.3% (13/16) of genes., Conclusions: The results of mutation analysis performed with a multi-gene panel and FFPE and fresh frozen tissue were highly concordant in patients with CRC, at both the variant and gene levels. There were, however, some important differences in mutation results between the two tissue types. Therefore, fresh frozen tissue should not routinely be replaced with FFPE tissue for mutation analysis with a multi-gene panel. Rather, FFPE tissue is a reasonable alternative for fresh frozen tissue when the latter is unavailable.;,citation_author=Xian Hua Gao;,citation_author=Juan Li;,citation_author=Hai Feng Gong;,citation_author=Guan Yu Yu;,citation_author=Peng Liu;,citation_author=Li Qiang Hao;,citation_author=Lian Jie Liu;,citation_author=Chen Guang Bai;,citation_author=Wei Zhang;,citation_publication_date=2020-03-13;,citation_cover_date=2020-03-13;,citation_year=2020;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7083147/;,citation_doi=10.3389/fonc.2020.00310;,citation_issn=2234-943X;,citation_pmid=32232001;,citation_volume=10;,citation_journal_title=Frontiers in Oncology;,citation_journal_abbrev=Front Oncol;">
<meta name="citation_reference" content="citation_title=Spatial mapping reveals human adipocyte subpopulations with distinct sensitivities to insulin;,citation_abstract=The contribution of cellular heterogeneity and architecture to white adipose tissue (WAT) function is poorly understood. Herein, we combined spatially resolved transcriptional profiling with single-cell RNA sequencing and image analyses to map human WAT composition and structure. This identified 18 cell classes with unique propensities to form spatially organized homo- and heterotypic clusters. Of these, three constituted mature adipocytes that were similar in size, but distinct in their spatial arrangements and transcriptional profiles. Based on marker genes, we termed these AdipoLEP, AdipoPLIN, and AdipoSAA. We confirmed, in independent datasets, that their respective gene profiles associated differently with both adipocyte and whole-body insulin sensitivity. Corroborating our observations, insulin stimulation in&nbsp;vivo by hyperinsulinemic-euglycemic clamp showed that only AdipoPLIN displayed a transcriptional response to insulin. Altogether, by mining this multimodal resource we identify that human WAT is composed of three classes of mature adipocytes, only one of which is insulin responsive.;,citation_author=Jesper Bäckdahl;,citation_author=Lovisa Franzén;,citation_author=Lucas Massier;,citation_author=Qian Li;,citation_author=Jutta Jalkanen;,citation_author=Hui Gao;,citation_author=Alma Andersson;,citation_author=Nayanika Bhalla;,citation_author=Anders Thorell;,citation_author=Mikael Rydén;,citation_author=Patrik L. Ståhl;,citation_author=Niklas Mejhert;,citation_publication_date=2021-09-07;,citation_cover_date=2021-09-07;,citation_year=2021;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S1550413121003636;,citation_issue=9;,citation_doi=10.1016/j.cmet.2021.07.018;,citation_issn=1550-4131;,citation_volume=33;,citation_language=en-US;,citation_journal_title=Cell Metabolism;,citation_journal_abbrev=Cell Metabolism;">
<meta name="citation_reference" content="citation_title=Integrative spatial analysis of cell morphologies and transcriptional states with MUSE;,citation_abstract=Spatial transcriptomics enables the simultaneous measurement of morphological features and transcriptional profiles of the same cells or regions in tissues. Here we present multi-modal structured embedding (MUSE), an approach to characterize cells and tissue regions by integrating morphological and spatially resolved transcriptional data. We demonstrate that MUSE can discover tissue subpopulations missed by either modality as well as compensate for modality-specific noise. We apply MUSE to diverse datasets containing spatial transcriptomics (seqFISH+, STARmap or Visium) and imaging (hematoxylin and eosin or fluorescence microscopy) modalities. MUSE identified biologically meaningful tissue subpopulations and stereotyped spatial patterning in healthy brain cortex and intestinal tissues. In diseased tissues, MUSE revealed gene biomarkers for proximity to tumor region and heterogeneity of amyloid precursor protein processing across Alzheimer brain regions. MUSE enables the integration of multi-modal data to provide insights into the states, functions and organization of cells in complex biological tissues.;,citation_author=Feng Bao;,citation_author=Yue Deng;,citation_author=Sen Wan;,citation_author=Susan Q. Shen;,citation_author=Bo Wang;,citation_author=Qionghai Dai;,citation_author=Steven J. Altschuler;,citation_author=Lani F. Wu;,citation_publication_date=2022-08;,citation_cover_date=2022-08;,citation_year=2022;,citation_fulltext_html_url=https://www.nature.com/articles/s41587-022-01251-z;,citation_issue=8;,citation_doi=10.1038/s41587-022-01251-z;,citation_issn=1546-1696;,citation_volume=40;,citation_language=en-US;,citation_journal_title=Nature Biotechnology;,citation_journal_abbrev=Nat Biotechnol;">
<meta name="citation_reference" content="citation_title=Biobanking of fresh frozen tissue: RNA is stable in nonfixed surgical specimens;,citation_abstract=Molecular tools for tissue profiling, such as expression microarrays and real-time PCR, generally require collection of fresh frozen tissues as sources of high-quality RNA. The fragile nature of RNA prompted us to examine the effects of storage time and transport conditions with regard to RNA integrity and gene expression in nonfixed surgical human specimens. At surgery, fresh normal tonsil and colon tissue was cut into pieces and snap frozen. Additional fresh tissue pieces were (i) left at room temperature, (ii) kept on ice, (iii) in normal saline or (iv) in a commercial RNA-stabilizing buffer (RNAlater) and snap frozen after 0.5, 1, 3, 6 and 16 h. Structural RNA integrity was analysed by microchip electrophoresis. Surprisingly, RNA remained stable in both tissue types under all conditions tested for up to 6–16 h. Gene expression by real-time PCR of cfos, HIF1α, Bcl2, PCNA, TGFβ1 and SMAD7 was analysed at different storage time points in tonsil tissue. Expression levels were essentially stable when samples were kept on ice, while marked regulation of single genes was observed during storage at room temperature, in normal saline and in RNAlater. Furthermore, we analysed selected tissue types from the local biobank representing 47 normal and malignant tissues transported on ice for up to 2–3 h before biobanking. RNA prepared from 45 of the 47 samples exhibited distinct ribosomal peaks indicating intact RNA. This study shows that RNA degradation is a minor problem during handling of fresh human tissue before biobanking. Our data indicate that nonfixed tissue specimens may be transported on ice for hours without any major influence on RNA quality and expression of the selected genes. However, further studies are warranted to clarify the impact of transport logistics on global gene expression.;,citation_author=Patrick Micke;,citation_author=Mitsuhiro Ohshima;,citation_author=Simin Tahmasebpoor;,citation_author=Zhi-Ping Ren;,citation_author=Arne Östman;,citation_author=Fredrik Pontén;,citation_author=Johan Botling;,citation_publication_date=2006-02-01;,citation_cover_date=2006-02-01;,citation_year=2006;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0023683722600205;,citation_issue=2;,citation_doi=10.1038/labinvest.3700372;,citation_issn=0023-6837;,citation_volume=86;,citation_language=en-US;,citation_journal_title=Laboratory Investigation;,citation_journal_abbrev=Laboratory Investigation;">
<meta name="citation_reference" content="citation_title=Single-cell analysis of human adipose tissue identifies depot- and disease-specific cell types;,citation_author=Jinchu Vijay;,citation_author=Marie-Frédérique Gauthier;,citation_author=Rebecca L. Biswell;,citation_author=Daniel A. Louiselle;,citation_author=Jeffrey J. Johnston;,citation_author=Warren A. Cheung;,citation_author=Bradley Belden;,citation_author=Albena Pramatarova;,citation_author=Laurent Biertho;,citation_author=Margaret Gibson;,citation_author=Marie-Michelle Simon;,citation_author=Haig Djambazian;,citation_author=Alfredo Staffa;,citation_author=Guillaume Bourque;,citation_author=Anita Laitinen;,citation_author=Johanna Nystedt;,citation_author=Marie-Claude Vohl;,citation_author=Jason D. Fraser;,citation_author=Tomi Pastinen;,citation_author=André Tchernof;,citation_author=Elin Grundberg;,citation_publication_date=2019-12-23;,citation_cover_date=2019-12-23;,citation_year=2019;,citation_fulltext_html_url=https://www.nature.com/articles/s42255-019-0152-6;,citation_issue=1;,citation_doi=10.1038/s42255-019-0152-6;,citation_issn=2522-5812;,citation_volume=2;,citation_language=en-US;,citation_journal_title=Nature Metabolism;">
<meta name="citation_reference" content="citation_title=Use of single-cell -omic technologies to study the gastrointestinal tract and diseases, from single cell identities to patient features;,citation_abstract=Single cells are the building blocks of tissue systems that determine organ phenotypes, behaviors, and functions. Understanding the differences between cell types and their activities might provide us with insights into normal tissue physiology, development of disease, and new therapeutic strategies. Although -omic level single-cell technologies are a relatively recent development that have been used only in research settings, these approaches might eventually be used in the clinic. We review the prospects of applying single-cell genome, transcriptome, epigenome, proteome, and metabolome analyses to gastroenterology and hepatology research. Combining data from multi-omic platforms coupled to rapid technological development could lead to new diagnostic, prognostic, and therapeutic approaches.;,citation_author=Mirazul Islam;,citation_author=Bob Chen;,citation_author=Jeffrey M. Spraggins;,citation_author=Ryan T. Kelly;,citation_author=Ken S. Lau;,citation_publication_date=2020-08-01;,citation_cover_date=2020-08-01;,citation_year=2020;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0016508520306569;,citation_issue=2;,citation_doi=10.1053/j.gastro.2020.04.073;,citation_issn=0016-5085;,citation_volume=159;,citation_language=en-US;,citation_journal_title=Gastroenterology;,citation_journal_abbrev=Gastroenterology;">
<meta name="citation_reference" content="citation_title=HuBMAP  formalin fixation and paraffin embedding of tissue samples v1;,citation_abstract=This method details formalin fixation and paraffin embedding of the HuBMAP tissue specimens. Protocol adopted from Marda Jorgensen, Jerelyn Nick (02/24/2020). HuBMAP: Paraffin Embedding Tissue Samples . https://dx.doi.org/10.17504/protocols.io.bam9ic96;,citation_author=Kelley Knizner;,citation_author=Christopher Simmons;,citation_publication_date=2020-12-14;,citation_cover_date=2020-12-14;,citation_year=2020;,citation_fulltext_html_url=https://www.protocols.io/view/hubmap-formalin-fixation-and-paraffin-embedding-of-bqp6mvre;,citation_doi=10.17504/protocols.io.bqp6mvre;">
<meta name="citation_reference" content="citation_title=Single-cell omics: Overview, analysis, and application in biomedical science;,citation_abstract=Single-cell sequencing methods provide the highest resolution insight into cellular heterogeneity. Owing to their rapid growth and decreasing cost, they are now widely accessible to scientists worldwide. Single-cell technologies enable analysis of a large number of cells, making them powerful tools to characterise rare cell types and refine our understanding of diverse cell states. Moreover, single-cell application in biomedical sciences helps to unravel mechanisms related to disease pathogenesis and outcome. In this Viewpoint, we briefly describe existing single-cell methods (genomics, transcriptomics, epigenomics, proteomics, and mulitomics), comment on available analysis tools, and give examples of method applications in the biomedical field.;,citation_author=Catarina M. Stein;,citation_author=Ralf Weiskirchen;,citation_author=Frederik Damm;,citation_author=Paulina M. Strzelecka;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1002/jcb.30134;,citation_issue=11;,citation_doi=10.1002/jcb.30134;,citation_issn=1097-4644;,citation_volume=122;,citation_language=en-US;,citation_journal_title=Journal of Cellular Biochemistry;">
<meta name="citation_reference" content="citation_title=Install TensorFlow and PyTorch with CUDA, cUDNN, and GPU Support in 3 Easy Steps;,citation_author=undefined Alex Watson;,citation_publication_date=2023-03-06;,citation_cover_date=2023-03-06;,citation_year=2023;,citation_fulltext_html_url=https://gretel.ai/blog/install-tensorflow-with-cuda-cdnn-and-gpu-support-in-4-easy-steps;,citation_language=en;">
<meta name="citation_reference" content="citation_title=Methyl alcohol safety;,citation_author=Flinn Scientific;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_fulltext_html_url=https://www.flinnsci.com/api/library/Download/762ec945eb5744518e71a014f92818cb;,citation_publisher=Flinn Scientific;">
<meta name="citation_reference" content="citation_title=Protocol for making a 4% formaldehyde solution in PBS. R&amp;amp;amp;d systems;,citation_abstract=Read our Protocol for Making a 4% Formaldehyde Solution in PBS.;,citation_author=R&amp;amp;D Systems;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.rndsystems.com/resources/protocols/protocol-making-4-formaldehyde-solution-pbs;,citation_language=en-US;">
<meta name="citation_reference" content="citation_title=Guide to laboratory sink/sewer disposal of wastes. Vanderbilt university medical center;,citation_author=Office of Clinical and Research Safety;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.vumc.org/safety/waste/chemical-waste-sewer-disposal;">
<meta name="citation_reference" content="citation_title=Methanol - safety data sheet;,citation_author=Regulatory Affairs;,citation_publication_date=2021-12-24;,citation_cover_date=2021-12-24;,citation_year=2021;,citation_fulltext_html_url=https://www.fishersci.com/store/msds?partNumber=A4124&amp;amp;amp;productDescription=methanol-certified-acs-fisher-chemical-10&amp;language=en&amp;countryCode=US;,citation_publisher=ThermoFisher Scientific;">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Research Reference</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./about.html" aria-current="page"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://github.com/pranavmishra90/research-reference"> 
<span class="menu-text">GitHub</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./about.html">About</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./about.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">About</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Data Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
 <span class="menu-text">Statistics</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data-analysis/statistics/anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ANOVA in Python</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Deploy</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./deploy/readme.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Datalad with GitHub and GIN siblings</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Lab Materials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./lab-materials/safe-disposal/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Safe disposal of lab chemicals</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab-materials/safe-disposal/methanol.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Methanol</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">Key Resources Table</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab-materials/key-resources-table/key-resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Key Resources Tables</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="true">
 <span class="menu-text">Nextgen Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./nextgen-analysis/lung-atlas-tutorial/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Exploring single-cell RNA sequencing using COVID-19 lung data</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false">
 <span class="menu-text">Platform</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./nextgen-analysis/platform/analysis-on-debian.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preparing your linux computer for next-generation -omics experiments</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="true">
 <span class="menu-text">Protocols</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./protocols/molecular-biology/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Molecular Biology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./protocols/molecular-biology/western-blot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Western Blot Protocol</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" aria-expanded="false">
 <span class="menu-text">PCR</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-11" class="collapse list-unstyled sidebar-section depth3 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./protocols/molecular-biology/PCR/RNA-isolation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">mRNA Isolation</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./protocols/recipes/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Recipes</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-12" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-13" aria-expanded="false">
 <span class="menu-text">Buffers</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-13" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-13" class="collapse list-unstyled sidebar-section depth3 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./protocols/recipes/buffers/tris-buffered-saline-TBS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Recipe: Tris-Buffered Saline (TBS)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./protocols/recipes/buffers/tris-buffered-saline-Tween-20.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tris-Buffered Saline with Tween 20 (TBST)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./protocols/recipes/buffers/tris-glycine-SDS-running-buffer.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tris-Glycine SDS Running Buffer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./protocols/recipes/buffers/tris-glycine-transfer-buffer.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tris-Glycine Transfer Buffer</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-14" aria-expanded="false">
 <span class="menu-text">Solvents</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-14" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-14" class="collapse list-unstyled sidebar-section depth3 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./protocols/recipes/solvents/formalin (10 percent).html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10% Formalin (4% Paraformaldehyde)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./protocols/recipes/solvents/pbs-phosphate-buffered-saline.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Phosphate Buffered Saline (PBS)</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-15" aria-expanded="false">
 <span class="menu-text">Working Stock</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-15" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-15" class="collapse list-unstyled sidebar-section depth3 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./protocols/recipes/working-stock/hydrochloric acid (1M).html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Hydrochloric Acid (1M)</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-16" aria-expanded="true">
 <span class="menu-text">Tutorial</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-16" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-16" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Tutorial/tables.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Adding tables easily to this site</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-17" aria-expanded="false">
 <span class="menu-text">Templates</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-17" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-17" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Tutorial/templates/recipe-template.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Recipe: Template</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#a-new-site-under-active-development" id="toc-a-new-site-under-active-development" class="nav-link active" data-scroll-target="#a-new-site-under-active-development">A new site under active development</a></li>
  <li><a href="#general-information" id="toc-general-information" class="nav-link" data-scroll-target="#general-information">General information</a>
  <ul class="collapse">
  <li><a href="#datalad-datasets-and-how-to-use-them" id="toc-datalad-datasets-and-how-to-use-them" class="nav-link" data-scroll-target="#datalad-datasets-and-how-to-use-them">DataLad datasets and how to use them</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/pranavmishra90/research-reference/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/pranavmishra90/research-reference/edit/main/about.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">About</h1>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Authors</div>
  <div class="quarto-title-meta-heading">Affiliations</div>
  
  </div>

<div class="quarto-title-meta">

      
  
    
  </div>
  


</header>


<section id="a-new-site-under-active-development" class="level2">
<h2 class="anchored" data-anchor-id="a-new-site-under-active-development">A new site under active development</h2>
<p>This is open-access site for research protocols, files, and techniques which I use. It is in an extreme infancy phase and is undergoing active development.</p>
<p>Thanks for visiting</p>
<p>Pranav</p>
</section>
<section id="general-information" class="level2">
<h2 class="anchored" data-anchor-id="general-information">General information</h2>
<p>This is a DataLad dataset (id: 3d6beff7-a55d-43a2-ab44-6def6a79bde5).</p>
<section id="datalad-datasets-and-how-to-use-them" class="level3">
<h3 class="anchored" data-anchor-id="datalad-datasets-and-how-to-use-them">DataLad datasets and how to use them</h3>
<p>This repository is a <a href="https://www.datalad.org/">DataLad</a> dataset. It provides fine-grained data access down to the level of individual files, and allows for tracking future updates. In order to use this repository for data retrieval, <a href="https://www.datalad.org/">DataLad</a> is required. It is a free and open source command line tool, available for all major operating systems, and builds up on Git and <a href="https://git-annex.branchable.com/">git-annex</a> to allow sharing, synchronizing, and version controlling collections of large files.</p>
<p>More information on how to install DataLad and <a href="http://handbook.datalad.org/en/latest/intro/installation.html">how to install</a> it can be found in the <a href="https://handbook.datalad.org/en/latest/index.html">DataLad Handbook</a>.</p>
<section id="get-the-dataset" class="level4">
<h4 class="anchored" data-anchor-id="get-the-dataset">Get the dataset</h4>
<p>A DataLad dataset can be <code>cloned</code> by running</p>
<pre><code>datalad clone &lt;url&gt;</code></pre>
<p>Once a dataset is cloned, it is a light-weight directory on your local machine. At this point, it contains only small metadata and information on the identity of the files in the dataset, but not actual <em>content</em> of the (sometimes large) data files.</p>
</section>
<section id="retrieve-dataset-content" class="level4">
<h4 class="anchored" data-anchor-id="retrieve-dataset-content">Retrieve dataset content</h4>
<p>After cloning a dataset, you can retrieve file contents by running</p>
<pre><code>datalad get &lt;path/to/directory/or/file&gt;</code></pre>
<p>This command will trigger a download of the files, directories, or subdatasets you have specified.</p>
<p>DataLad datasets can contain other datasets, so called <em>subdatasets</em>. If you clone the top-level dataset, subdatasets do not yet contain metadata and information on the identity of files, but appear to be empty directories. In order to retrieve file availability metadata in subdatasets, run</p>
<pre><code>datalad get -n &lt;path/to/subdataset&gt;</code></pre>
<p>Afterwards, you can browse the retrieved metadata to find out about subdataset contents, and retrieve individual files with <code>datalad get</code>. If you use <code>datalad get &lt;path/to/subdataset&gt;</code>, all contents of the subdataset will be downloaded at once.</p>
</section>
<section id="stay-up-to-date" class="level4">
<h4 class="anchored" data-anchor-id="stay-up-to-date">Stay up-to-date</h4>
<p>DataLad datasets can be updated. The command <code>datalad update</code> will <em>fetch</em> updates and store them on a different branch (by default <code>remotes/origin/master</code>). Running</p>
<pre><code>datalad update --merge</code></pre>
<p>will <em>pull</em> available updates and integrate them in one go.</p>
</section>
<section id="find-out-what-has-been-done" class="level4">
<h4 class="anchored" data-anchor-id="find-out-what-has-been-done">Find out what has been done</h4>
<p>DataLad datasets contain their history in the <code>git log</code>. By running <code>git log</code> (or a tool that displays Git history) in the dataset or on specific files, you can find out what has been done to the dataset or to individual files by whom, and when.</p>


</section>
</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{untitled,
  author = {},
  title = {About},
  url = {ref.drpranavmishra.com/about.html},
  langid = {en-US}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-untitled" class="csl-entry quarto-appendix-citeas" role="listitem">
<div class="">About. <a href="https://ref.drpranavmishra.com/about.html">ref.drpranavmishra.com/about.html</a></div>
</div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>Created by Pranav Kumar Mishra || License CC-BY-SA</p>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/pranavmishra90/research-reference/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li><li><a href="https://github.com/pranavmishra90/research-reference/edit/main/about.qmd" class="toc-action"><i class="bi empty"></i>Edit this page</a></li></ul></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>